### PR TITLE
feat(policy): allow provider config to specify to latest policy version

### PR DESCRIPTION
**Pull request by idilhaq**

### CODE REVIEW

File: core/appeal/service.go [hunk 1]
```diff
@@ -850,14 +851,24 @@ func (s *Service) getPoliciesMap(ctx context.Context) (map[string]map[uint]*doma
...
 		id := p.ID
 		version := p.Version
 		if policiesMap[id] == nil {
 			policiesMap[id] = map[uint]*domain.Policy{}
 		}
 		policiesMap[id][version] = p
+		if i == 0 {
+			// set the map 0 as the latest version
+			policiesMap[id][0] = p
+		}
```
Comment (rahmatrhd) [inline core/appeal/service.go:871]: please modify the logic under the assumption that policy IDs also can be multiple. 
Example: (`{policy_id}@{version}`)
- `test-policy@1`
- `test-policy@2`
- `other-policy@1`
- `other-policy@2`
- `other-policy@3`

Comment (idilhaq) [reply core/appeal/service.go]: @rahmatrhd updated the flow based on the cases above ya

File: core/appeal/service_test.go [hunk 2]
```diff
@@ -998,6 +1001,451 @@ func (s *ServiceTestSuite) TestCreate() {
...
+				URN:  "provider1",
+				Config: &domain.ProviderConfig{
+					Appeal: &domain.AppealConfig{
+						AllowPermanentAccess:         true,
+						AllowActiveAccessExtensionIn: "24h",
+					},
+					Resources: []*domain.ResourceConfig{
+						{
+							Type: "resource_type_1",
+							Policy: &domain.PolicyConfig{
```
Comment (rahmatrhd) [inline core/appeal/service_test.go:1040]:
```suggestion
							Policy: &domain.PolicyConfig{ // specify policy with version
```

File: core/appeal/service_test.go [hunk 3]
```diff
@@ -998,6 +1001,451 @@ func (s *ServiceTestSuite) TestCreate() {
...
+							Roles: []*domain.Role{
+								{
+									ID:          "role_id",
+									Permissions: []interface{}{"test-permission-1"},
+								},
+							},
+						},
+						{
+							Type: "resource_type_2",
+							Policy: &domain.PolicyConfig{
```
Comment (rahmatrhd) [inline core/appeal/service_test.go:1053]:
```suggestion
							Policy: &domain.PolicyConfig{ // specify policy without version (always use latest)
```

File: core/appeal/service_test.go [hunk 4]
```diff
@@ -998,6 +1001,451 @@ func (s *ServiceTestSuite) TestCreate() {
...
+					},
+				},
+				Description: "The answer is 42",
+			},
+			{
+				ID:            "2",
+				ResourceID:    "2",
+				Resource:      resources[1],
+				PolicyID:      "policy_2",
+				PolicyVersion: 20,
```
Comment (rahmatrhd) [inline core/appeal/service_test.go:1339]:
```suggestion
				PolicyVersion: 20, // result expected to be created with the latest policy
```

**Review by rahmatrhd — APPROVED** [review]

**Pull request merged**